crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -227,6 +227,28 @@ def openstackbackup
     }, status: :unprocessable_entity
   end
 
+  def mode
+    if request.post?
+      Api::Upgrade.upgrade_mode=params[:mode]
+      render json: {}, status: :ok
+    else
+      render json: {
+        mode: Api::Upgrade.upgrade_mode
+      }
+    end
+  rescue ::Crowbar::Error::SaveUpgradeModeError,
+         ::Crowbar::Error::SaveUpgradeStatusError,
+         ::Crowbar::Error::UpgradeError => e
+    render json: {
+      errors: {
+        mode: {
+          data: e.message,
+          help: "Please refer to the error message in the response."
```

Comment:
I think we should just drop the `help` part if it does not include anything interesting